### PR TITLE
symbols: Consolidate spawning of ctags

### PR DESCRIPTION
**Pull request by chrismwendt**
Previously, we were creating the parser 2 different ways.

This will make it easier to replace ctags with tree-sitter for select languages in the future to address https://github.com/sourcegraph/customer/issues/976#issuecomment-1146451111

## Test plan

Go tests

### CODE REVIEW

**Review by keegancsmith — APPROVED** [review]

File: cmd/symbols/parser/parser.go [hunk 1]
```diff
@@ -245,3 +248,23 @@ func shouldPersistEntry(e *ctags.Entry) bool {
...
+func SpawnCtags(ctagsConfig types.CtagsConfig) (ctags.Parser, error) {
+	options := ctags.Options{
+		Bin:                ctagsConfig.Command,
+		PatternLengthLimit: ctagsConfig.PatternLengthLimit,
+	}
+	if ctagsConfig.LogErrors {
+		options.Info = log.New(os.Stderr, "ctags: ", log.LstdFlags)
+	}
+	if ctagsConfig.DebugLogs {
+		options.Debug = log.New(os.Stderr, "DBUG ctags: ", log.LstdFlags)
```
Comment (chrismwendt) [inline cmd/symbols/parser/parser.go:261]: CleanShot 2022-06-06 at 17 43 52

https://buildkite.com/sourcegraph/sourcegraph/builds/152324

ctags only takes the `log.Logger` struct, which is banned. I'm removing the logging to get past CI for now.

This check was added in https://github.com/sourcegraph/sourcegraph/pull/34855 cc @bobheadxi

Comment (bobheadxi) [reply cmd/symbols/parser/parser.go]: If there's no way around it you can add a documented exception here: https://github.com/sourcegraph/sourcegraph/blob/main/dev/sg/linters/liblog.go

Comment (chrismwendt) [reply cmd/symbols/parser/parser.go:261]: Added an exception instead 👍

Comment (bobheadxi) [reply cmd/symbols/parser/parser.go:261]: Something to consider: https://github.com/sourcegraph/sourcegraph/pull/36687

**Pull request merged**